Lib/traceback.py

```diff
@@ -1292,7 +1292,10 @@ def _format_syntax_error(self, stype, **kwargs):
                 yield '    {}\n'.format(ltext)
             else:
                 offset = self.offset
-                end_offset = self.end_offset if self.end_offset not in {None, 0} else offset
+                if self.lineno == self.end_lineno:
+                    end_offset = self.end_offset if self.end_offset not in {None, 0} else offset
+                else:
+                    end_offset = len(self.text)
```

Comment:
```suggestion
                    end_offset = len(self.text) + 1
```
?

Without this, issue's traceback looks like
```pycon
>>> dummy_call(
...     "dummy value"
...     foo="bar",
... )
... 
  File "<unknown>", line 2
    "dummy value"
    ^^^^^^^^^^^^
SyntaxError: invalid syntax. Perhaps you forgot a comma?
```
I think all line should be marked, till the end (where comma shoule be).

Reply:
I consider it but it looks weird to highlight white space. Notice this path will be taken for multi line syntax errors so it's not just about missing commas.

Reply:
In this particular case it's not a white space.  Also, your solution doesn't avoid such scenario.  Just add two white spaces:)

Probably, that part - another issue: trailing spaces shouldn't be included in exception's metadata.

Reply:
Oh apologies, I think I misread your comment! You are absolutely right. Let me amend this in a slightly different way to merge it with a code path that's after this

Reply:
Hummmm, this is what I get when testing in my machine with the current code:

```
./python lel.py
  File "/home/pablogsal/github/python/main/lel.py", line 2
    "dummy value"
    ^^^^^^^^^^^^^
SyntaxError: invalid syntax. Perhaps you forgot a comma?
```

and this is with your suggestion:

```
./python lel.py
  File "/home/pablogsal/github/python/main/lel.py", line 2
    "dummy value"
    ^^^^^^^^^^^^^^
SyntaxError: invalid syntax. Perhaps you forgot a comma?
```

Reply:
Am I missing something?

Reply:
Ah, seems that when executing over a file there is an extra newline character in the text metadata!